test/playback-watcher.test.js

```diff
@@ -742,6 +743,293 @@ QUnit.test('jumps to buffered content if seeking just before', function(assert)
+    this.player.dispose();
+  }
+});
+
+loaderTypes.forEach(function(type) {
+  QUnit.test(`detects ${type} appends without buffer changes and excludes`, function(assert) {
+    this.setup();
+    const loader = this.mpc[`${type}SegmentLoader_`];
+    const track = {label: 'foobar', mode: 'showing'};
+
+    if (type === 'subtitle') {
+      loader.track = () => track;
+      sinon.stub(this.player.tech_.textTracks(), 'removeTrack');
+    }
+
+    this.setBuffered(videojs.createTimeRanges([[0, 30]]));
+    loader.trigger('updateend');
+    assert.equal(this.playbackWatcher[`${type}StalledDownloads_`], 0, '1st append 0 stalled downloads');
+    assert.deepEqual(this.usageEvents, {}, 'no usage events');
+
+    loader.trigger('updateend');
+    assert.equal(this.playbackWatcher[`${type}StalledDownloads_`], 1, '2nd append 1 stalled downloads');
+    assert.deepEqual(this.usageEvents, {}, 'no usage events');
+
+    loader.trigger('updateend');
+    assert.equal(this.playbackWatcher[`${type}StalledDownloads_`], 2, '3rd append 2 stalled downloads');
+    assert.deepEqual(this.usageEvents, {}, 'no usage events');
+
+    loader.trigger('updateend');
+    assert.equal(this.playbackWatcher[`${type}StalledDownloads_`], 0, '4th append 0 stalled downloads');
+
+    const expectedUsage = {};
+
+    expectedUsage[`vhs-${type}-download-exclusion`] = 1;
+
+    if (type !== 'subtitle') {
+      expectedUsage['hls-rendition-blacklisted'] = 1;
+    }
+
+    assert.deepEqual(this.usageEvents, expectedUsage, 'usage as expected');
+
+    if (type !== 'subtitle') {
+      const message = 'Playback cannot continue. No available working or supported playlists.';
+
+      assert.equal(this.mpcErrors, 1, 'one mpc error');
+      assert.equal(this.mpc.error, message, 'mpc error set');
+      assert.equal(this.player.error().message, message, 'player error set');
+      assert.equal(this.env.log.error.callCount, 1, 'player error logged');
+      assert.equal(this.env.log.error.args[0][1], message, 'error message as expected');
+
+      this.env.log.error.resetHistory();
+    } else {
+      const message = 'Text track "foobar" is not working correctly. It will be disabled and excluded.';
+
+      assert.equal(this.mpcErrors, 0, 'no mpc error set');
+      assert.notOk(this.player.error(), 'no player error set');
+      assert.equal(this.player.textTracks().removeTrack.callCount, 1, 'text track remove called');
+      assert.equal(this.player.textTracks().removeTrack.args[0][0], track, 'text track remove called with expected');
+      assert.equal(track.mode, 'disabled', 'mode set to disabled now');
+      assert.equal(this.env.log.warn.callCount, 1, 'warning logged');
+      assert.equal(this.env.log.warn.args[0][0], message, 'warning message as expected');
+
+      this.env.log.warn.resetHistory();
+    }
+  });
+
+  if (type !== 'subtitle') {
+    QUnit.test(`detects ${type} appends without buffer changes and excludes many playlists`, function(assert) {
```

Comment:
Since this only covers main and audio, and uses a conditional, I think it would be better to have two separate tests.

Reply:
If we decide to do different excluding with audio loader we can do that, but right now this would just end in a copy paste of a test.